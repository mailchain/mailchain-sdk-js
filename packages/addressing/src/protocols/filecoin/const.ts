import { SECP256K1PrivateKey } from '@mailchain/crypto';
import { decodeHex } from '@mailchain/encoding';

// Alice
export const AliceFilStr = 'f410faiikgixcdd7alrsbpunjlen4etkcarbfbghn7wi';
export const AliceFilEthAddressStr = '0x0210a322e218fe05c6417d1a9591bc24d4204425';
export const AliceFilEthPrivateKeyStr = '6c115551accbd74b46401a531af1dd0890034c18137dddab52a53c626e75b051';
export const AliceFilEthPrivateKey = new SECP256K1PrivateKey(decodeHex(AliceFilEthPrivateKeyStr));

// Bob
export const BobFilAddressStr = 'f410f5blfdi7tmto6o2aa2up4vwtdm7qcnaiqpeazuwy';
export const BobFilEthAddressStr = '0xe85651a3f364dde76800d51fcada6367e0268110';
export const BobFilEthPrivateKeyStr = '2c6b29d5545e5804b7fb7e65404782e57dddea84cbcec9b2586add6403808965';
export const BobFilEthPrivateKey = new SECP256K1PrivateKey(decodeHex(BobFilEthPrivateKeyStr));

export enum FilPrefix {
	F4ETHEREUM = 'f410',
	T4ETHEREUM = 't410',
}
