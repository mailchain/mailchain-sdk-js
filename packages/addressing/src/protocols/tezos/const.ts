export enum Prefix {
	TZ1 = 'tz1',
	TZ2 = 'tz2',
	TZ3 = 'tz3',
	KT = 'KT',
	KT1 = 'KT1',

	EDSK2 = 'edsk2',
	SPSK = 'spsk',
	P2SK = 'p2sk',

	EDPK = 'edpk',
	SPPK = 'sppk',
	P2PK = 'p2pk',

	EDESK = 'edesk',
	SPESK = 'spesk',
	P2ESK = 'p2esk',

	EDSK = 'edsk',
	EDSIG = 'edsig',
	SPSIG = 'spsig',
	P2SIG = 'p2sig',
	SIG = 'sig',
	EXPR = 'expr',
}

export const prefix = {
	[Prefix.TZ1]: new Uint8Array([6, 161, 159]),
	[Prefix.TZ2]: new Uint8Array([6, 161, 161]),
	[Prefix.TZ3]: new Uint8Array([6, 161, 164]),
	[Prefix.KT]: new Uint8Array([2, 90, 121]),
	[Prefix.KT1]: new Uint8Array([2, 90, 121]),

	[Prefix.EDSK]: new Uint8Array([43, 246, 78, 7]),
	[Prefix.EDSK2]: new Uint8Array([13, 15, 58, 7]),
	[Prefix.SPSK]: new Uint8Array([17, 162, 224, 201]),
	[Prefix.P2SK]: new Uint8Array([16, 81, 238, 189]),

	[Prefix.EDPK]: new Uint8Array([13, 15, 37, 217]),
	[Prefix.SPPK]: new Uint8Array([3, 254, 226, 86]),
	[Prefix.P2PK]: new Uint8Array([3, 178, 139, 127]),

	[Prefix.EDESK]: new Uint8Array([7, 90, 60, 179, 41]),
	[Prefix.SPESK]: new Uint8Array([0x09, 0xed, 0xf1, 0xae, 0x96]),
	[Prefix.P2ESK]: new Uint8Array([0x09, 0x30, 0x39, 0x73, 0xab]),

	[Prefix.EDSIG]: new Uint8Array([9, 245, 205, 134, 18]),
	[Prefix.SPSIG]: new Uint8Array([13, 115, 101, 19, 63]),
	[Prefix.P2SIG]: new Uint8Array([54, 240, 44, 52]),
	[Prefix.SIG]: new Uint8Array([4, 130, 43]),
	[Prefix.EXPR]: new Uint8Array([13, 44, 64, 27]),
};
