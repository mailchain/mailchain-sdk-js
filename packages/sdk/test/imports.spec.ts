import path from 'path';
import { getImportsFromDir } from '../../../../collect-imports';
import pckg from '../package.json';
import { explicitExports } from '../rollup.const';

describe('check imports snapshot', () => {
	const imports = [
		...new Set(
			getImportsFromDir(path.resolve(`${__dirname}`, '..'), ['.ts', '.js']).filter((it) =>
				it.startsWith(pckg.name),
			),
		),
	].sort();
	test('compare imports', () => {
		expect(imports).toMatchSnapshot('imports');
	});
	test('compare explicit imports', () => {
		expect(explicitExports).toMatchSnapshot('imports_explicit');
	});
});
