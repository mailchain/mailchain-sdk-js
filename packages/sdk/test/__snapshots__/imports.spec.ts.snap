// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check imports snapshot compare explicit imports: imports_explicit 1`] = `
[
  "@mailchain/sdk/internal/",
]
`;

exports[`check imports snapshot compare imports: imports 1`] = `
[
  "@mailchain/sdk",
]
`;
